vendor/engines/saml_authentication/db/migrate/20180413194031_add_saml_session_index_to_user.rb

```diff
@@ -0,0 +1,6 @@
+class AddSamlSessionIndexToUser < ActiveRecord::Migration
+  def change
+    add_column :users, :saml_session_index, :string
```

Comment:
I'm confused by what we need this for...

Reply:
I don't exactly understand the details, but according to https://github.com/apokalipto/devise_saml_authenticatable "This stores the session index defined by the IDP during login.  If provided it will be used as a salt for the user's session to facilitate an IDP initiated logout request." It kind of seems like it's not fully fleshed out (they don't even tell you it needs to be a database column), but I think the idea is if you initiate the logout from the IdP, this is used to help log you out.

I might play around a little more with it to see if we can get away without it. My suspicion is that we would need to do some more overriding of the devise saml gem.